internal/credential/options.go

```diff
@@ -0,0 +1,39 @@
+package credential
+
+import (
+	"github.com/hashicorp/boundary/internal/util/template"
+)
+
+// GetOpts - iterate the inbound Options and return a struct
+func GetOpts(opt ...Option) (*options, error) {
+	opts := getDefaultOptions()
+	for _, o := range opt {
+		if o == nil {
+			continue
+		}
+		if err := o(opts); err != nil {
+			return nil, err
+		}
+	}
+	return opts, nil
+}
+
+// Option - how Options are passed as arguments.
+type Option func(*options) error
+
+// options = how options are represented
+type options struct {
+	WithTemplateData template.Data
```

Comment:
This is a pattern we have all over TBH. I realize it seems odd but since we re-parse options each time having it be read-only isn't terribly important.